src/Behat/Config/Config.php

```diff
@@ -11,6 +11,13 @@ public function __construct(
     ) {
     }
 
+    public function withProfile(ProfileInterface $profile): self
+    {
+        $this->settings[$profile->name()] = $profile->toArray();
```

Comment:
What if they have already defined a profile with that name? Should we perhaps throw rather than silently overwriting (bearing in mind we're in the context of a single config here, obviously imports are merged together later).

The examples in the feature files already suggest that the PHP config files might become a lot more verbose / have more variables etc than the YAML ones - I think that increases the risk of accidentally having a duplicate key, and I wonder if we should protect against that?

Reply:
We could add an optional arguments `bool $overwrite = false`
But cause you are able to redefine the same profile configuration via another file, it will not work.

Reply:
No you cannot. Defining the same key twice in a Yaml config file is a Yaml syntax error.

Reply:
I mean, in multiple yaml configuration files,  you are able to redefine a suite with same key as defined in another file.